ejb3/src/main/java/org/jboss/as/ejb3/remote/protocol/versionone/XidTransactionPrepareTask.java

```diff
@@ -91,11 +91,10 @@ private int prepareTransaction() throws Throwable {
             // check the recovery store - it's possible that the "prepare" is coming in as part of recovery operation and the subordinate
             // tx may not yet be in memory, but might be in the recovery store
             final Transaction recoveredTransaction = tryRecoveryForImportedTransaction();
-            // still not found, so just return
+            // still not found
             if (recoveredTransaction == null) {
-                if (EjbLogger.EJB3_INVOCATION_LOGGER.isDebugEnabled()) {
```

Comment:
@kurobako I think it is a good idea. The fix for tx ctx propagation with async calls is alredy in a PR, in case you want to test locally with the commit, here it is:
https://github.com/wildfly/wildfly/pull/9462/